src/mappings/ethereumDIDRegistry.template.ts

```diff
@@ -0,0 +1,39 @@
+import { json, {{#ipfs}}ipfs, {{/ipfs}}Bytes, JSONValueKind } from '@graphprotocol/graph-ts'
+import { DIDAttributeChanged } from '../types/EthereumDIDRegistry/EthereumDIDRegistry'
+
+import { addQm, createOrLoadGraphAccount } from './helpers'
+import { jsonToString } from './utils'
+
+export function handleDIDAttributeChanged(event: DIDAttributeChanged): void {
+  let id = event.params.identity.toHexString()
+  let graphAccount = createOrLoadGraphAccount(id, event.params.identity, event.block.timestamp)
+  // OFFCHAIN_DATANAME = keccak256("GRAPH NAME SERVICE")
+  // 0x72abcb436eed911d1b6046bbe645c235ec3767c842eb1005a6da9326c2347e4c
+  if (
+    event.params.name.toHexString() ==
+    '0x72abcb436eed911d1b6046bbe645c235ec3767c842eb1005a6da9326c2347e4c'
+  ) {
+    graphAccount.metadataHash = event.params.value
+    {{#ipfs}}
+    // TODO optimization - make this more robust, since value is BYTES not BYTES32, and if someone
+    // called it directly, it could crash the subgraph
+    let hexHash = addQm(event.params.value) as Bytes
+    let base58Hash = hexHash.toBase58()
+
+    let ipfsData = ipfs.cat(base58Hash)
+    if (ipfsData != null) {
+      let data = json.fromBytes(ipfsData as Bytes).toObject()
+      graphAccount.codeRepository = jsonToString(data.get('codeRepository'))
+      graphAccount.description = jsonToString(data.get('description'))
+      graphAccount.image = jsonToString(data.get('image'))
+      graphAccount.displayName = jsonToString(data.get('displayName'))
+      let isOrganization = data.get('isOrganization')
+      if (isOrganization != null && isOrganization.kind === JSONValueKind.BOOL) {
+        graphAccount.isOrganization = isOrganization.toBool()
+      }
+      graphAccount.website = jsonToString(data.get('website'))
+    }
+    {{/ipfs}}
```

Comment:
I would do this differently. What I would do is add a file called `ipfsHelpers.ts.template` and put the following functions in there:

```
export function fetchGraphAccountMetadata(graphAccount: GraphAccount, ipfsHash: Bytes): void {
  {{#ipfs}}
  ...
  {{/ipfs}}
  // otherwise do nothing
}

export fetchSubgraphMetadata(subgraph: Subgraph, ipfsHash: Bytes): void {
  {{#ipfs}}
  ...
  {{/ipfs}}
  graphAccount.description = ''
  ...
}
```

This way the templating is limited to only the IPFS-related code and the `gns.ts` mapping is unaffected. Feels cleaner and more isolated to me.